### PR TITLE
[WIP] Dev server mode, base.js refactor

### PR DESCRIPTION
**Pull request by pascalopitz**
- Adds `-dev` flag so dev on JS files can happen with just a reload, instead of re-building artifact
- Rips out JQuery and moment from `base.js`
- Fixes nil pointer exception in template
- Fixes wrong type exception in template

### CODE REVIEW

**Comment by pascalopitz** [comment]
@buro9 I added a dev mode toggle in this to enable faster process when editing JS files. What do you think?

File: web/server/static/js/base.js [hunk 1]
```diff
@@ -1,130 +1,174 @@
...
+	const relativeFormatter = new Intl.RelativeTimeFormat(undefined, {
+		numeric: 'auto'
+	})
+
+	const DIVISIONS = [
+		{ amount: 60, name: 'seconds' },
+		{ amount: 60, name: 'minutes' },
+		{ amount: 24, name: 'hours' },
+		{ amount: 7, name: 'days' },
+		{ amount: 4.34524, name: 'weeks' },
```
Comment (buro9) [inline web/server/static/js/base.js:12]: Where did 4.34524 come from?

It looks like `(((365*3+366)/4)/12) / (((365*3+366)/4)/52)` to determine weeks in a month over a year but it's not quite, as that produces 4.3 recurring.

File: web/server/server.go [hunk 2]
```diff
@@ -206,16 +208,29 @@ func staticFiles(version string) http.Handler {
...
 
 			next.ServeHTTP(w, r)
 		})
 	})
 
-	// Serve static files
-	router.Mount(`/`,
-		http.FileServer(http.FS(inlinedFiles)),
-	)
+	if *opts.IsDevelopment {
```
Comment (buro9) [inline web/server/server.go:221]: This is nice... I like this for the simplicity of it.

**Review by buro9 — APPROVED** [review]
I'm happy to merge... it works right?

I've approved, you can hit merge I believe.

File: web/server/static/js/base.js (hunk 1 above)
Comment (pascalopitz) [reply web/server/static/js/base.js:12]: Literally from the article I linked in the comment.
Googling, the same factor for month to week conversion seems to be widely used, i.e.:

https://w3percentagecalculator.com/how-many-weeks-in-a-month/
https://www.checkyourmath.com/convert/time/weeks_months.php
https://www.asknumbers.com/DateTimeConversion.aspx

I have no idea where they get the number from and if it's more or less accurate than your formula.

Comment (buro9) [reply web/server/static/js/base.js:12]: LOL, I should read the comment :)

I'm happy to trust the number if the entire world does... I just pause when I see a magic number and see if I can understand it. I think it's roughly what my formula was but probably accounts for it better (i.e. the 4.3 recurring is likely too approximate and someone's stabbed around at the decimal places to make it "good enough" without being too approximate). I cannot find the source of the number used, but hey it works for me.

**Pull request merged**